packages/flutter/lib/src/material/data_table.dart

```diff
@@ -1051,6 +1070,54 @@ class DataTable extends StatelessWidget {
   }
 }
 
+/// Height settings for DataRows.
+/// Either a fixed row height or a row height that adjusts to its content can be specified.
+@immutable
+class DataRowHeightStyle with Diagnosticable {
+  /// Creates height settings where all rows are created with the same fixed height.
+  const DataRowHeightStyle.fixed({
+    required double height
+  }) : fixedHeight = height, verticalPadding = 0.0;
+
+  /// Creates height settings where the row height adjusts to the content.
+  const DataRowHeightStyle.auto({
+    required this.verticalPadding,
```

Comment:
> This seems a bit unnecessary. Could the user not just wrap the cell content in Padding?

Yes. I like the suggestion.

> Is there not a way we can determine the row needs to be bigger without changing the user facing code?

The first commit used a different approach than the current code, it changed the meaning of the `dataRowHeight` parameter to `minDataRowHeight` and used it in BoxConstraints as `constraints: BoxConstraints(minHeight: minDataRowHeight),` (see https://github.com/flutter/flutter/pull/102886/commits/77c06b0f3398f1ecab4defef5e5ee97fb3b0a50a)
The first commit also had an additional parameter `topBottomRowPadding` that seemed necessary because rows with one line appeared to have padding because the row height was much larger than the content but for rows with multiple lines the `BoxConstraints(minHeight: minDataRowHeight)` would apply and there was no padding. Without needing to support padding the original solution might be simplified.
So one solution would be to use the BoxConstraints approach again but without the additional topBottomRowPadding parameter. However, then I would suggest to rename `dataRowHeight` to `minDataRowHeight` because that is what it would represent. Using BoxConstraints instead of a fixed height would probably be more expensive to render but it can be tested if that really makes a difference. 

One note about your concerns for changing user facing code. I suppose, for most users there is no depreciation warning because most users use the default height and do not explicitly set dataRowHeight. The users that have set dataRowHeight will get a depreciation warning. However I suppose most apps do not use a lot of DataTable widgets, so there are not many places that need to be adjusted.